hapi-utils/src/main/java/com/hedera/services/throttles/GasLimitBucketThrottle.java

```diff
@@ -0,0 +1,47 @@
+package com.hedera.services.throttles;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+public class GasLimitBucketThrottle {
+
+    private final DiscreteLeakyBucket bucket;
+    private static final long ONE_SECOND_IN_NANOSECONDS = 1_000_000_000;
+
+    public GasLimitBucketThrottle(long capacity) {
+        this.bucket = new DiscreteLeakyBucket(capacity);
+    }
+
+    public boolean allow(long txGasLimit, long elapsedNanos) {
+
+        double toLeak = ((double) elapsedNanos / ONE_SECOND_IN_NANOSECONDS) * bucket.totalCapacity();
```

Comment:
Since this is being used by `DeterministicThrottle`, it must not do floating-point arithmetic.